Artesian/Artesian.SDK/Service/Clients/Client.cs

```diff
@@ -145,24 +146,40 @@ public Client(IArtesianServiceConfig config, string Url, ArtesianPolicyConfig po
 
                         if (!res.IsSuccessStatusCode)
                         {
-                            var responseText = await res.Content.ReadAsStringAsync();
-                            var exceptionMessage = $"Failed handling REST call to WebInterface {method} {_url + resource}. Returned status: {res.StatusCode}. Content: \n";
+                            ArtesianSdkProblemDetail problemDetail = null;
+                            string responseText = string.Empty;
 
-                            if (res.StatusCode == HttpStatusCode.BadRequest)
+                            if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
                             {
-                                var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
-                                responseText = _tryDecodeText(responseDeserialized);
-
-                                throw new ArtesianSdkValidationException(exceptionMessage + responseText);
+                                problemDetail = await res.Content.ReadAsAsync<ArtesianSdkProblemDetail>(_formatters, ctk);
+                            }
+                            else
+                            {
+                                if (res.StatusCode == HttpStatusCode.BadRequest)
+                                {
+                                    responseText = _tryDecodeText(await res.Content.ReadAsAsync<ArtesianSdkProblemDetail>(_formatters, ctk));
```

Comment:
I'm sure the original implementation wasn't deserializing into ArtesianSdkProblemDetail but something else, maybe <dynamic>. please check the original usage of _tryDecodeText

Reply:
Reverted to  <object>